etcd-manager/pkg/volumes/aws/volumes.go

```diff
@@ -98,16 +99,29 @@ func NewAWSVolumes(clusterName string, volumeTags []string, nameTag string) (*AW
 		return nil, fmt.Errorf("error querying ec2 metadata service (for az): %v", err)
 	}
 
-	a.instanceID, err = a.metadata.GetMetadata("instance-id")
+	hostname, err := os.Hostname()
```

Comment:
I think getting this from metadata would be more reliable.

Reply:
I debated this a bit ... metadata has no notion of a primary IPv6 address. So which one would be the right one in the metadata. Always assume index 0's address? lowest index if index 0 has no IP? In CCM there is some dead code for ipv6 from metadata and it is not that nice.

Reply:
Meant the hostname.

Reply:
If we want to get rid of hostname override for RBN in kOps, we need to rely on hostname working. If not, kubelet/kube-proxy will also fail at this.

Reply:
Yes, but os.Hostname ran in a container doesn't seem like the best thing. Using metadata makes is OS independent.

Reply:
That's a good point. It works for kOps because of hostnetworking and such. Without that, this will probably fail. But then, metadata is maybe not available either.

Right now, there is a flag for passing in listen address, but it is ignored, and etcd-manager will always look up. But yeah, I can switch this to looking up this from metadata.